File: LeetCode_Record/2309.兼具大小写的最好英文字母.cpp
#include <string>
using namespace std;

class Solution {
public:
    string greatestLetter(string s) {
        int letters[58] = {0};
        for(char& c : s){
            letters[c - 'A']++;
        }
        
        for(int i = 25; i >= 0; i--){
            if(letters[i] > 0 && letters[i + 32] > 0){
                return string(1, i + 'A');
            }
        }
        return "";
    }
};

File: LeetCode_Record/2309.兼具大小写的最好英文字母.java
class Solution {
    public String greatestLetter(String s) {
        int[] letters = new int[58];
        int n = s.length();
        for(int i = 0; i < n; i++){
            letters[s.charAt(i) - 'A']++;
        }
        
        for(int i = 25; i >= 0; i--){
            if(letters[i] > 0 && letters[i + 32] > 0){
                return String.valueOf((char)(i + 'A'));
            }
        }
        return "";
    }
}